### ISSUE TITLE
ECS provider panic

### ISSUE THREAD

**Issue by tskinn** [containous/traefik#1140]
### What version of Traefik are you using (`traefik version`)?
v1.2.0-rc1

### What is your environment & configuration (arguments, toml...)?
[web]
address = ":8080"
[ecs]
Cluster = "staging"
Watch = true
RefreshSeconds = 15
ExposedByDefault = false
Region = "us-east-1"

### What did you do?
Scaled the amount of tasks running that have containers with these labels:
traefik.frontend.rule=Host:test.traefik.io
traefik.enable=true

### What did you expect to see?
I expected to see the new tasks to be registered in traefik and for traefik to route traffic to them.

### What did you see instead?
```time="2017-02-08T19:12:08Z" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference"
goroutine 81 [running]:
runtime/debug.Stack(0x4e808f, 0xc42008e240, 0x1d9a271)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x79
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/containous/traefik/safe.defaultRecoverGoroutine(0x1ada8e0, 0xc420012030)
	/go/src/github.com/containous/traefik/safe/routine.go:147 +0x7d
github.com/containous/traefik/safe.OperationWithRecover.func1.1(0xc4203d3e60)
	/go/src/github.com/containous/traefik/safe/routine.go:155 +0x56
panic(0x1ada8e0, 0xc420012030)
	/usr/local/go/src/runtime/panic.go:458 +0x243
text/template.errRecover(0xc4203d3a50)
	/usr/local/go/src/text/template/exec.go:140 +0x2ad
panic(0x1ada8e0, 0xc420012030)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/containous/traefik/provider.(*ecsInstance).Host(0xc4207bb3c8, 0x0, 0x0)
	<autogenerated>:38 +0x7b
reflect.Value.call(0x1c17300, 0xc4207bb3c8, 0x613, 0x1d78247, 0x4, 0x2b8cdd8, 0x0, 0x0, 0x5e9c03, 0x2b6bba0, ...)
	/usr/local/go/src/reflect/value.go:434 +0x5c8
reflect.Value.Call(0x1c17300, 0xc4207bb3c8, 0x613, 0x2b8cdd8, 0x0, 0x0, 0xc420028fb8, 0x7, 0x7)
	/usr/local/go/src/reflect/value.go:302 +0xa4
text/template.(*state).evalCall(0xc4203d39d0, 0x1d0b1a0, 0xc4207bb3c8, 0x199, 0x1c17300, 0xc4207bb3c8, 0x613, 0x2a85ea0, 0xc4208d2630, 0xc42046e087, ...)
	/usr/local/go/src/text/template/exec.go:658 +0x530
text/template.(*state).evalField(0xc4203d39d0, 0x1d0b1a0, 0xc4207bb3c8, 0x199, 0xc42046e087, 0x4, 0x2a85ea0, 0xc4208d2630, 0xc420012dc0, 0x1, ...)
	/usr/local/go/src/text/template/exec.go:549 +0xdb6
text/template.(*state).evalFieldChain(0xc4203d39d0, 0x1d0b1a0, 0xc4207bb3c8, 0x199, 0x1d0b1a0, 0xc4207bb3c8, 0x199, 0x2a85ea0, 0xc4208d2630, 0xc420012db0, ...)
	/usr/local/go/src/text/template/exec.go:520 +0x22f
text/template.(*state).evalFieldNode(0xc4203d39d0, 0x1d0b1a0, 0xc4207bb3c8, 0x199, 0xc4208d2630, 0xc420012dc0, 0x1, 0x1, 0x0, 0x0, ...)
	/usr/local/go/src/text/template/exec.go:484 +0x114
text/template.(*state).evalCommand(0xc4203d39d0, 0x1d0b1a0, 0xc4207bb3c8, 0x199, 0xc4208d2600, 0x0, 0x0, 0x0, 0xc4203d3568, 0x78ac3f, ...)
	/usr/local/go/src/text/template/exec.go:422 +0x866
text/template.(*state).evalPipeline(0xc4203d39d0, 0x1d0b1a0, 0xc4207bb3c8, 0x199, 0xc42069be50, 0x0, 0x0, 0x98)
	/usr/local/go/src/text/template/exec.go:400 +0xee
text/template.(*state).walk(0xc4203d39d0, 0x1d0b1a0, 0xc4207bb3c8, 0x199, 0x2a85c60, 0xc4208d2660)
	/usr/local/go/src/text/template/exec.go:226 +0x423
text/template.(*state).walk(0xc4203d39d0, 0x1d0b1a0, 0xc4207bb3c8, 0x199, 0x2a85fc0, 0xc4208d22d0)
	/usr/local/go/src/text/template/exec.go:234 +0x13d
text/template.(*state).walkRange.func1(0x19a9060, 0xc4200130f0, 0x82, 0x1d0b1a0, 0xc4207bb3c8, 0x199)
	/usr/local/go/src/text/template/exec.go:321 +0x132
text/template.(*state).walkRange(0xc4203d39d0, 0x1b06920, 0xc42051f260, 0x99, 0xc4209ff8c0)
	/usr/local/go/src/text/template/exec.go:330 +0x300
text/template.(*state).walk(0xc4203d39d0, 0x1b06920, 0xc42051f260, 0x99, 0x2a86140, 0xc4209ff8c0)
	/usr/local/go/src/text/template/exec.go:237 +0x59c
text/template.(*state).walk(0xc4203d39d0, 0x1b06920, 0xc42051f260, 0x99, 0x2a85fc0, 0xc4208d2210)
	/usr/local/go/src/text/template/exec.go:234 +0x13d
text/template.(*Template).execute(0xc4209ff7c0, 0x2a67e80, 0xc4201e0070, 0x1b06920, 0xc42051f260, 0x0, 0x0)
	/usr/local/go/src/text/template/exec.go:189 +0x204
text/template.(*Template).Execute(0xc4209ff7c0, 0x2a67e80, 0xc4201e0070, 0x1b06920, 0xc42051f260, 0x0, 0x0)
	/usr/local/go/src/text/template/exec.go:175 +0x53
github.com/containous/traefik/provider.(*BaseProvider).getConfiguration(0xc42017eea0, 0x1d91332, 0x12, 0xc42053f5c0, 0x1b06920, 0xc42051f260, 0x20, 0x0, 0x0)
	/go/src/github.com/containous/traefik/provider/provider.go:90 +0x561
github.com/containous/traefik/provider.(*ECS).loadECSConfig(0xc42017eea0, 0x2a804c0, 0xc42008ea00, 0xc4209ceab0, 0x1e930b5, 0x1e930b4, 0xc41ffd12f5)
	/go/src/github.com/containous/traefik/provider/ecs.go:189 +0x3f4
github.com/containous/traefik/provider.(*ECS).Provide.func2.2(0x0, 0x0)
	/go/src/github.com/containous/traefik/provider/ecs.go:121 +0xc4
github.com/containous/traefik/safe.OperationWithRecover.func1(0x0, 0x0)
	/go/src/github.com/containous/traefik/safe/routine.go:159 +0x75
github.com/containous/traefik/vendor/github.com/cenk/backoff.RetryNotify(0xc4203fe370, 0x2a73a80, 0xc4203fe380, 0x1e96e28, 0xc4203fe360, 0x0)
	/go/src/github.com/containous/traefik/vendor/github.com/cenk/backoff/retry.go:32 +0x3f
github.com/containous/traefik/provider.(*ECS).Provide.func2(0xc420444620)
	/go/src/github.com/containous/traefik/provider/ecs.go:158 +0x1d2
github.com/containous/traefik/safe.(*Pool).Go.func1()
	/go/src/github.com/containous/traefik/safe/routine.go:77 +0x3a
github.com/containous/traefik/safe.GoWithRecover.func1(0x1e96ec8, 0xc4204ee000)
	/go/src/github.com/containous/traefik/safe/routine.go:141 +0x4d
created by github.com/containous/traefik/safe.GoWithRecover
	/go/src/github.com/containous/traefik/safe/routine.go:142 +0x49
time="2017-02-08T19:12:08Z" level=error msg="ECS connection error Panic in operation: %!s(<nil>), retrying in 1.105608358s"```

**Comment by emilevauge** [comment]
Ping @lpetre :'(

**Comment by lpetre** [comment]
Yikes I'll take a look asap

**Comment by lpetre** [comment]
Attempting to address this here: https://github.com/containous/traefik/pull/1143

@emilevauge the panic is happing on this line:
https://github.com/containous/traefik/blob/master/provider/ecs.go#L381

Unfortunately the AWS Go SDK uses pointers for absolutely everything, and isn't really clear on when to expect nil values. I've added a bunch of additional checking and filtering, so it shouldn't be possible to get this panic in the future.  Let me know what you think.

**Comment by lpetre** [comment]
Hi @emilevauge, this is fixed by PR #1143. @tskinn tested it locally. Let me know if you need anything done to get it merged.
